### ISSUE TITLE
[Decompiler (C++)] Maximum number of operands is 30

### ISSUE THREAD

**Issue by lab313ru** [NationalSecurityAgency/ghidra#3250]
Decompiler allows to have maximum 30 operands:
```cpp
VarnodeData invar[30];
```

Here it's hardcoded:
https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc#L899

And here:
https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/decompile/cpp/translate.cc#L989

Related to (where maximum is 8): https://github.com/NationalSecurityAgency/ghidra/issues/3243

**Comment by astrelsky** [comment]
@lab313ru may you try the following to check if it helps whatever you're working on?

```git
git remote add astrelsky https://github.com/astrelsky/ghidra.git
git fetch astrelsky
git cherry-pick df99a3200f3ccf92cb882a79c37d0177ad181aeb
git cherry-pick 9858f51764c4e39ba263908a3e0b73c795c963a5
```

**Comment by lab313ru** [comment]
You can check it by yourself too. Just change some opcode implementation to `pcodeop blabla;` and `blabla(1:4, 2:4, 3:4, 4:4, 5:4, 6:4, 7:4, 8:4, 9:4, 10:4, 11:4, 12:4, 13:4, 14:4, 15:4, 16:4, 17:4, 18:4, 19:4, 20:4, 21:4, 22:4, 23:4, 24:4, 25:4, 26:4, 27:4, 28:4, 29:4, 30:4, 31:4);`

**Comment by astrelsky** [comment]
> You can check it by yourself too. Just change some opcode implementation to `pcodeop blabla;` and `blabla(1:4, 2:4, 3:4, 4:4, 5:4, 6:4, 7:4, 8:4, 9:4, 10:4, 11:4, 12:4, 13:4, 14:4, 15:4, 16:4, 17:4, 18:4, 19:4, 20:4, 21:4, 22:4, 23:4, 24:4, 25:4, 26:4, 27:4, 28:4, 29:4, 30:4, 31:4);`

It does work as testing that is no different from any other architecture. What I wouldn't see would be any performance issues if they occur.

**Comment by lab313ru** [comment]
So, should I do only those two cherry-picks, right?

**Comment by lab313ru** [comment]
What about Java part?

**Comment by astrelsky** [comment]
> What about Java part?

Both fixes are there. You can just pull the two branches that are named #issue as I removed everything else.

It will also let you keep working on what you are implementing.

**Comment by Swyter** [comment]
Man, if the NSA research directorate is hiring I know the right candidate. Thanks a lot for your work, Andrew.

**Comment by astrelsky** [comment]
> Man, if the NSA research directorate is hiring I know the right candidate. Thanks a lot for your work, Andrew.

You're welcome. Unfortunately I'm no longer looking for work and am happy where I am at the moment. You never know what the future holds though.

**Comment by caheckman** [comment]
Addressed by 79c3508f54389a5bf54090871614831d2f96caa6
